runtime/j9vm/jvm.c

```diff
@@ -3989,15 +3989,79 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 			UDATA slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
 			Trc_SC_LoadLibrary_OpenShared(libName);
 
-/* jdk17+ calls JVM_LoadLibrary with decorated library names. If the following is done
- * then it overwrites the real error with a failure to load "liblib<name>.so.so".
- */
-#if JAVA_SPEC_VERSION < 17
 			if (0 != slOpenResult) {
-				slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags | J9PORT_SLOPEN_DECORATE);
-				Trc_SC_LoadLibrary_OpenShared_Decorate(libName);
+				char *libNameNotDecorated = (char *)libName;
+#if JAVA_SPEC_VERSION >= 17
+				/* JDK17+ jdk.internal.loader.NativeLibraries.load() calls JVM_LoadLibrary() with a decorated library name,
+				 * i.e., a path to the library name returned from Java_java_lang_System_mapLibraryName(nameNoPrefixNoExtension).
+				 * The library name passed to j9sl_open_shared_library() with the flag J9PORT_SLOPEN_DECORATE
+				 * has to be platform independent, i.e., it must not contain any prefix or file extension.
+				 */
+				char libPath[EsMaxPath] = {0};
+				const char *fileExt = strrchr(libName, '.');
+				BOOLEAN doOpenLibrary = TRUE;
+				if (NULL != fileExt) {
+					const size_t libSuffixLength = strlen(fileExt);
+					const char *fileNameTmp = strrchr(libName, DIR_SEPARATOR);
+					const char *fileName = (NULL == fileNameTmp) ? libName : fileNameTmp + 1;
+#if defined(J9OS_I5) || defined(WIN32)
+					/* no library prefix for J9OS_I5 and WIN32 */
+					const size_t libStrLength = 0;
+#else /* defined(J9OS_I5) || defined(WIN32) */
+					/*  strlen("lib") = 3 */
+					const size_t libStrLength = 3;
+					if (0 != strncmp("lib", fileName, libStrLength)) {
+						/* A decorated library name is expected to start with lib prefix for platforms other than WIN32 & J9OS_I5.
+						 * pass to j9sl_open_shared_library w/o modification.
```

Comment:
Please fold, and start the each sentence with a capital.

Reply:
Updated